src/utils/commonUtils.js

```diff
@@ -72,4 +73,61 @@ module.exports = {
 
     return fileContents;
   },
+
+  getCurrentMetadata() {
+    return {
+      osInfo: {
+        arch: os.arch(),
+        hostname: os.hostname(),
+        platform: os.platform(),
+        release: os.release(),
+        type: os.type(),
+        username: os.userInfo().username,
+      },
+      processInfo: {
+        execPath: process.execPath,
+        memoryUsage: process.memoryUsage(),
+        uptime: process.uptime(),
+        version: process.version,
+        versions: process.versions,
+      },
+      version: require('../../package.json').version,
```

Comment:
put the require up at the top to take advantage of node module caching

Reply:
I put it here to make sure it gets the latest version every time getCurrentMetadata is called. Not sure if it matters in practice since the collector should be restarted any time the version is updated.

Reply:
OK either way